kedro/framework/cli/starters.py

```diff
@@ -22,22 +24,53 @@
     KedroCliError,
     _clean_pycache,
     _filter_deprecation_warnings,
+    _safe_load_entry_point,
     command_with_verbosity,
+    get_entry_points,
 )
 
 KEDRO_PATH = Path(kedro.__file__).parent
 TEMPLATE_PATH = KEDRO_PATH / "templates" / "project"
-
-_STARTER_ALIASES = {
-    "astro-airflow-iris",
-    "standalone-datacatalog",
-    "pandas-iris",
-    "pyspark",
-    "pyspark-iris",
-    "spaceflights",
-}
 _STARTERS_REPO = "git+https://github.com/kedro-org/kedro-starters.git"
 
+
+@define(order=True)
+class KedroStarterSpec:  # pylint: disable=too-few-public-methods
+    """Specification of custom kedro starter template
+    Args:
+        alias: alias of the starter which shows up on `kedro starter list` and `kedro new`
+        template_path: the path to git repository
+        directory: by default it will look for `alias`, optionally override with custom directory.
```

Comment:
```suggestion
        directory: optional directory inside the repository where the starter resides.
```
(copied from `DIRECTORY_ARG_HELP`)